apps/src/gamelab/AnimationTab/ListItemThumbnail.jsx

```diff
@@ -43,14 +42,34 @@ var staticStyles = {
 /**
  * Animation or Frame thumbnail.
  */
-var ListItemThumbnail = React.createClass({
+const ListItemThumbnail = React.createClass({
   propTypes: {
+    animation: React.PropTypes.shape(METADATA_SHAPE).isRequired,
     index: React.PropTypes.number,
-    isSelected: React.PropTypes.bool,
-    src: React.PropTypes.string.isRequired
+    isSelected: React.PropTypes.bool
+  },
+
+  getInitialState() {
+    return {
+      previewSize: 0
+    };
+  },
+
+  componentDidMount() {
+    this.recalculatePreviewSize();
+  },
+
+  forceResize() {
+    this.recalculatePreviewSize();
+  },
```

Comment:
Wondering if we want to come up with a convention to indicate which methods might get called externally. One (common) approach would be to do like we do elsewhere and put _ on methods that *arent* called externally, i.e. `recalculatePreviewSize` would be `recalculatePreviewSize_`. Or we could make the assumption that most React methods will not be called externally, and be good about adding jsdoc @public for those that are.